apps/remote_control/lib/lexical/remote_control/search/indexer/extractors/function_definition.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Lexical.RemoteControl.Search.Indexer.Extractors.FunctionDefinition do
+  alias Lexical.Ast.Analysis
+  alias Lexical.Document
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.Formats
+  alias Lexical.RemoteControl.Search.Indexer.Entry
+  alias Lexical.RemoteControl.Search.Indexer.Metadata
+  alias Lexical.RemoteControl.Search.Indexer.Source.Block
+  alias Lexical.RemoteControl.Search.Indexer.Source.Reducer
+
+  @function_definitions [:def, :defp]
+
+  def extract({definition, metadata, [{fn_name, _, args}, _body]} = ast, %Reducer{} = reducer)
+      when is_atom(fn_name) and definition in @function_definitions do
+    range = get_extent(reducer.analysis, metadata)
```

Comment:
I'm curious about the benefits of having `get_extent` instead of `get_range` here.

Reply:
I'm not following. Is this a naming concern?

Reply:
It's not a concern, my English isn't good. From what I understand, these two words are often interchangeable. I just want to know under what circumstances they would be interchangeable. Here both words appear, the variable is 'range', and the function name is 'get_extent'.

Reply:
this sounds like a naming concern. `get_range` sounded pretty generic and non-descriptive to me, so I picked `get_extent` though perhaps. `get_definition_range` will be nicer? I'm open to other names.

Reply:
> get_definition_range will be nicer?

👍🏻